app/controllers/agent/IncomeSourceController.scala

```
@@ -56,32 +58,32 @@ class IncomeSourceController @Inject()(val authService: AuthService, subscriptio
         formWithErrors =>
           Future.successful(BadRequest(view(incomeSourceForm = formWithErrors, isEditMode = isEditMode))),
         incomeSource => {
+          lazy val linearJourney: Future[Result] =
+            subscriptionDetailsService.saveIncomeSource(incomeSource) map { _ =>
+              incomeSource match {
+                case IncomeSourceModel(true, _, _) =>
+                  Redirect(controllers.agent.business.routes.BusinessNameController.show())
+                case IncomeSourceModel(_, true, _) =>
+                  Redirect(controllers.agent.business.routes.PropertyAccountingMethodController.show())
+                case IncomeSourceModel(_, _, true) =>
+                  Redirect(controllers.agent.business.routes.PropertyAccountingMethodController.show())
+                case _ =>
+                  Redirect(controllers.agent.routes.CheckYourAnswersController.show())
```

Should probably just throw an exception here instead of routing the user to the check your answers

---

done